include/ulight/impl/lang/xml.hpp

```diff
@@ -25,11 +27,23 @@ struct Name_Match_Result {
 /// beginning of str satisfies the predicate is_stop_sequence.
 /// Characters that are not allowed in a name e.g & and > are
 /// marked for later highlighting.
+template <auto is_stop>
 [[nodiscard]]
-Name_Match_Result match_name_permissive(
-    std::u8string_view str,
-    const std::function<bool(std::u8string_view)>& is_stop_sequence
-);
+Name_Match_Result match_name_permissive(const std::u8string_view str)
```

Comment:
```suggestion
template <std::invocable<std::u8string_view> Stop>
[[nodiscard]]
Name_Match_Result match_name_permissive(const std::u8string_view str, Stop is_stop)
```
There's no practical benefit to requiring to the stop function to be a template argument.

Reply:
When I added the required clause I got compiler errors not sure why tho.

Reply:
What's the error? Maybe it's just a missing `#include <type_traits>`.

Reply:
The error looks like this.
```
ulight/include/ulight/impl/lang/xml.hpp:33:1: error: expected primary-expression before ‘[’ token
   33 | [[nodiscard]]
      | ^
```